nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,265 @@
+                            dynamicPropertyMap.put(entry.getName(), resultSet);
+                        }
+                    }
+                }
+                Map<String, Object> parameters = new HashMap<>(dynamicPropertyMap);
+
+                parameters.putAll(input.getAttributes());
+
+                graphResponses.addAll(executeQuery(recordScript, parameters));
+
+                recordIndex++;
+                if (recordIndex % 100 == 0) {
+                    long now = System.currentTimeMillis();
+                    batchStart = System.currentTimeMillis();
+                }
+            }
+            writer.finishRecordSet();
+            long end = System.currentTimeMillis();
+            delta = (end - start) / 1000;
+            if (getLogger().isDebugEnabled()){
+                getLogger().debug(String.format("Took %s seconds.", delta));
+            }
+        } catch (Exception ex) {
+            getLogger().error("", ex);
+            failed = true;
+        } finally {
+            if (failed) {
+                session.remove(output);
+                session.getProvenanceReporter().route(input, REL_FAILURE);
+                session.transfer(input, REL_FAILURE);
+            } else {
+                input = session.putAttribute(input, GRAPH_OPERATION_TIME, String.valueOf(delta));
+                session.getProvenanceReporter().send(input, clientService.getTransitUrl(), delta*1000);
+                session.transfer(input, REL_SUCCESS);
+                FlowFile graph = session.create(input);
+                session.getProvenanceReporter().clone(input, graph);
+                boolean error = false;
+                try (OutputStream os = session.write(graph)) {
+                    String graphOutput = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(graphResponses);
+                    os.write(graphOutput.getBytes(StandardCharsets.UTF_8));
+                    session.getProvenanceReporter().modifyContent(graph);
+                } catch (Exception ex) {
+                    getLogger().error("", ex);
+                    error = true;
+                } finally {
+                    if (!error) {
+                        session.getProvenanceReporter().route(graph, REL_GRAPH);
+                        session.transfer(graph, REL_GRAPH);
+                    } else {
+                        session.remove(graph);
+                    }
+                }
+                if (errors.get() > 0) {
+                    session.getProvenanceReporter().route(output, REL_ERRORS);
+                    session.transfer(output, REL_ERRORS);
+                } else {
+                    session.remove(output);
+                }
+            }
+        }
+    }
+
+    private String getScript(PropertyDescriptor script, ProcessContext context, FlowFile input) {
```

Comment:
This method can be deleted and the one line we need moved to `onTrigger`.

Reply:
Removed